enterprise/cloud.oracle/src/org/netbeans/modules/cloud/oracle/compute/ClusterItem.java

```diff
@@ -34,5 +43,30 @@ public ClusterItem(OCID id, String compartmentId, String name, String tenancyId,
     public ClusterItem() {
         super();
     }
+
+    public String getConfig() {
+        return config;
+    }
+
+    public void setConfig(String config) {
+        this.config = config;
+    }
     
+    public void update() {
+        OCIProfile profile = OCIManager.getDefault().getActiveProfile(this);
+        ContainerEngineClient containerEngineClient = profile.newClient(ContainerEngineClient.class);
+
+        
+        CreateKubeconfigRequest request = CreateKubeconfigRequest.builder()
+                .clusterId(getKey().getValue())
+                .build();
+        
+        CreateKubeconfigResponse response = containerEngineClient.createKubeconfig(request);
+        
+        try {
+            setConfig(new String(response.getInputStream().readAllBytes(), Charset.defaultCharset()));
+        } catch (IOException ex) {
+            Exceptions.printStackTrace(ex);
```

Comment:
I'd suggest to somehow signal this to the ClusterNode so the user is informed (icon, state, ...)

Reply:
Will be addressed in subsequent PRs